app/routes.py

```diff
@@ -1,2 +1,257 @@
-from flask import Blueprint
+from flask import Blueprint,jsonify, request
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from sqlalchemy import asc, desc
+from datetime import datetime
+import os
+import requests
 
+
+# task and goal blueprints
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+# helper functions
+def task_dict(task):
+    return {
+        "id": task.task_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.completed_at is not None
+    }
+
+def task_with_goal_dict(task):
+    return {
+        "id": task.task_id,
+        "goal_id": task.goal_id,
+        "title": task.title,
+        "description": task.description,
+        "is_complete": task.completed_at is not None
+    }
```

Comment:
We can combine these two functions. Instead of writing a completely new function to return a version of the task dictionary with a goal_id, we can use a conditional to add the goal_id if it exists for a task:

```suggestion
    def task_dict(self): 
       task = {
            "id": self.task_id,
            "title": self.title,
            "description": self.description,
            "is_complete": self.completed_at is not None
        }
        
       if self.goal_id:
           task["goal_id"] = self.goal_id
       
       return task
```